src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -733,6 +718,12 @@ void GcsActorManager::OnActorCreationSuccess(const std::shared_ptr<GcsActor> &ac
         RAY_CHECK(!worker_id.IsNil());
         RAY_CHECK(!node_id.IsNil());
         RAY_CHECK(created_actors_[node_id].emplace(worker_id, actor_id).second);
+
+        if (!actor->IsDetached() && worker_client_factory_) {
+          // This actor is owned. Send a long polling request to the actor's
+          // owner to determine when the actor should be removed.
+          PollOwnerForActorOutOfScope(actor);
```

Comment:
In `PollOwnerForActorOutOfScope`, we should handle the case where the owner has died.

Reply:
fixed